### PR TITLE
Add pipeline support

### PR DESCRIPTION
**Pull request by patcadelina**
This PR is built on top of @oleg-nenashev's [work on JEP-200](https://github.com/jenkinsci/aws-device-farm-plugin/pull/3) and deprecates @snbuback's [work on Pipeline plugin support](https://github.com/jenkinsci/aws-device-farm-plugin/pull/2).

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
@patcadelina Looks good to me. In order to make the UX better, I would also recommend adding `@Symbol` to the Recorder's descriptor so that Jenkins can generate a nice-looking step without the `step` wrapper

Would you be also interested to take ownership of the plugin? @nikhil-dabhade does not respond to ping, and ownership transfer could be the way to go to deliver improvements and JEP-200 fixes

**Comment by patcadelina** [comment]
> In order to make the UX better, I would also recommend adding @Symbol to the Recorder's descriptor so that Jenkins can generate a nice-looking step without the step wrapper

Sure, will update this PR and run some tests.

> Would you be also interested to take ownership of the plugin? @nikhil-dabhade does not respond to ping, and ownership transfer could be the way to go to deliver improvements and JEP-200 fixes

I think I can help with housekeeping. Are you taking over the plugin?

**Comment by oleg-nenashev** [comment]
> I think I can help with housekeeping. Are you taking over the plugin?

I was about submitting ownership transfer request to release a new version. I do not use this plugin on my instances, so it would be difficult for me to be a maintainer (taking all other commitments into account). But I am happy to help with Jenkins expertise if needed.

I will start the thread in the developer mailing list

**Comment by patcadelina** [comment]
> I was about submitting ownership transfer request to release a new version. I do not use this plugin on my instances, so it would be difficult for me to be a maintainer (taking all other commitments into account). But I am happy to help with Jenkins expertise if needed.

Got it, let's go from there.

**Comment by oleg-nenashev** [comment]
@patcadelina what is your Jenkins JIRA ID?

**Comment by patcadelina** [comment]
should be @patcadelina

**Comment by oleg-nenashev** [comment]
I have pinged @nikhil-dabhade directly. If there is no response by tomorrow EoD, I will start the official ownership handover thread in the dev list

File: src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmTestResultAction.java
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmTestResultAction.java]: It can be refactored later to get rid of `hudson.model.Run<?, ?> owner` in API at all

**Comment by oleg-nenashev** [comment]
@patcadelina I am going to release #3 so that we have a clean swimlane for this PR and #4. Would it be possible to remerge this PR with Master so that my commits get removed from the diff?

**Comment by patcadelina** [comment]
Sure I'll do a rebase

**Comment by oleg-nenashev** [comment]
1.16 is out, so this PR can be merged once it's rebased/upmerged and once the tests pass

**Comment by patcadelina** [comment]
@oleg-nenashev this should be ready for another review

**Comment by morphysk** [comment]
any updates on this?

**Comment by thaiphv** [comment]
@oleg-nenashev do you plan to merge this pull request?

**Comment by oleg-nenashev** [comment]
I could probably release it if needed. I expected @patcadelina to merge and release it once he gets maintainer permissions, but he has not responded to my recent pings. So I can release it if needed, but I cannot commit on good maintenance if something goes wrong

**Comment by patcadelina** [comment]
Hey @oleg-nenashev, I'll work on releasing this. Can you comment/approve [repository-permissions-updater #793](https://github.com/jenkins-infra/repository-permissions-updater/pull/793)?

**Comment by oleg-nenashev** [comment]
@patcadelina cool, welcome back! I have merged your pull request

**Comment by patcadelina** [comment]
Sure thanks, will get this tested out internally before releasing.

**Pull request merged**
